google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -154,10 +154,10 @@ class TracingMessageBatch : public MessageBatch {
     return [oc = opentelemetry::context::RuntimeContext::GetCurrent(),
             next = child_->Flush(),
             spans = std::move(batch_sink_spans)](auto f) mutable {
-      internal::DetachOTelContext(oc);
```

Comment:
I'm not sure if this order matters or not, I think I moved it when experimenting to see what was wrong, but I think it makes sense to remove the otel context after the spans are added.

Reply:
s/added/ended/ but yeah. The order doesn't really matter. After might be slightly better, if it is convenient.

Reply:
SG